modules/sdk/src/main/scala/org/tessellation/sdk/domain/fork/ForkInfoEntries.scala

```diff
@@ -0,0 +1,35 @@
+package org.tessellation.sdk.domain.fork
+
+import cats.syntax.eq._
+import cats.syntax.option._
+
+import scala.collection.immutable.Queue
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.PosInt
+
+@derive(eqv, show)
+sealed case class ForkInfoEntries(maxSize: PosInt, value: Queue[ForkInfo]) {
+  def add(entry: ForkInfo): ForkInfoEntries = {
+    val updated = (if (value.size === maxSize) value.tail else value) :+ entry
+
+    ForkInfoEntries(maxSize, updated)
+  }
+
+  def getEntries: Iterable[ForkInfo] = value
+}
+
+object ForkInfoEntries {
+  def empty(maxSize: PosInt): ForkInfoEntries = new ForkInfoEntries(maxSize, Queue.empty)
+
+  def from(maxSize: PosInt, entries: IterableOnce[ForkInfo]): Option[ForkInfoEntries] =
+    if (entries.iterator.size > maxSize)
+      none
+    else
+      entries.iterator
+        .foldLeft(ForkInfoEntries.empty(maxSize)) { case (acc, entry) => acc.add(entry) }
+        .some
+}
```

Comment:
This doesn't address my concern of being able to create an invalid object. Here's an alternate implementation to consider:
```
sealed trait ForkInfoEntries {
  def add(entry: ForkInfo): ForkInfoEntries
  def getEntries: Iterable[ForkInfo]
}

object ForkInfoEntries {
  private case class ForkInfoEntriesImpl(maxSize: PosInt, value: Queue[ForkInfo]) extends ForkInfoEntries {
    def add(entry: ForkInfo): ForkInfoEntries = {
      val updated = (if (value.size === maxSize) value.tail else value) :+ entry

      copy(value = updated)
    }

    def getEntries: Iterable[ForkInfo] = value
  }

  def empty(maxSize: PosInt): ForkInfoEntries = ForkInfoEntriesImpl(maxSize, Queue.empty)

  def from(maxSize: PosInt, entries: Iterable[ForkInfo]): ForkInfoEntries =
    ForkInfoEntriesImpl(maxSize, Queue.from(entries.takeRight(maxSize)))
}
```

Reply:
Changed to

```
sealed trait ForkInfoEntries {
  def add(entry: ForkInfo): ForkInfoEntries
  def getEntries: Iterable[ForkInfo]
}

object ForkInfoEntries {
  implicit val eq: Eq[ForkInfoEntries] = Eq.instance[ForkInfoEntries] { (e1, e2) =>
    e1.getEntries.toSeq === e2.getEntries.toSeq
  }

  implicit val show: Show[ForkInfoEntries] = Show.show[ForkInfoEntries] { e =>
    e.getEntries.mkString("[", ",", "]")
  }

  private case class ForkInfoEntriesImpl(maxSize: PosInt, value: Queue[ForkInfo]) extends ForkInfoEntries {
    def add(entry: ForkInfo): ForkInfoEntries = {
      val updated = (if (value.size === maxSize) value.tail else value) :+ entry

      copy(value = updated)
    }

    def getEntries: Iterable[ForkInfo] = value
  }

  def empty(maxSize: PosInt): ForkInfoEntries = ForkInfoEntriesImpl(maxSize, Queue.empty)

  def from(maxSize: PosInt, entries: Iterable[ForkInfo]): ForkInfoEntries =
    ForkInfoEntriesImpl(maxSize, Queue.from(entries.takeRight(maxSize)))

}
```